apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -0,0 +1,203 @@
+    return this.props.tableColumns.filter(
+      column =>
+        records.filter(
+          record =>
+            this.isCellEmpty(record[column]) ||
+            typeof record[column] === 'number'
+        ).length === records.length
+    );
+  };
+
+  componentDidUpdate(previousProps, prevState) {
+    const visualizerJustOpened =
+      !prevState.isVisualizerOpen && this.state.isVisualizerOpen;
+
+    const recordsChangedWithVisualizerOpen =
+      this.props.tableRecords !== previousProps.tableRecords &&
+      this.state.isVisualizerOpen;
+
+    if (visualizerJustOpened || recordsChangedWithVisualizerOpen) {
+      let parsedRecords = this.parseRecords();
+      let numericColumns = this.findNumericColumns(parsedRecords);
+      this.setState({
+        parsedRecords: parsedRecords,
+        numericColumns: numericColumns
+      });
+    }
+  }
+
+  render() {
+    let options = this.props.tableColumns;
+    let disabledOptions = [];
+
+    const disableNonNumericColumns =
+      this.state.chartType === 'Scatter Plot' ||
+      this.state.chartType === 'Histogram';
+
+    if (disableNonNumericColumns) {
+      disabledOptions = _.difference(
+        this.props.tableColumns,
+        this.state.numericColumns
+      );
+    }
+
+    const chartOptions = (
+      <div>
+        <div style={rowStyle.container}>
+          <label style={rowStyle.description}>Chart Title</label>
+          <input
+            style={rowStyle.input}
+            value={this.state.chartTitle}
+            onChange={event => this.setState({chartTitle: event.target.value})}
+          />
+        </div>
+
+        <DropdownField
+          displayName="Chart Type"
+          options={['Bar Chart', 'Histogram', 'Cross Tab', 'Scatter Plot']}
+          value={this.state.chartType}
+          onChange={event => this.setState({chartType: event.target.value})}
+        />
+
+        {this.state.chartType === 'Histogram' && (
+          <div style={rowStyle.container}>
+            <label style={rowStyle.description}>Bucket Size</label>
+            <input
+              style={rowStyle.input}
+              value={this.state.bucketSize}
+              onChange={event =>
+                this.setState({bucketSize: event.target.value})
+              }
+            />
+          </div>
+        )}
+        <DropdownField
+          displayName="Values"
+          options={options}
+          disabledOptions={disabledOptions}
+          value={this.state.column1}
+          onChange={event => this.setState({column1: event.target.value})}
+        />
+
+        {(this.state.chartType === 'Cross Tab' ||
+          this.state.chartType === 'Scatter Plot') && (
+          <DropdownField
+            displayName="Y Values"
+            options={options}
+            disabledOptions={disabledOptions}
+            value={this.state.column2}
+            onChange={event => this.setState({column2: event.target.value})}
+          />
+        )}
+      </div>
+    );
```

Comment:
The value of `chartOptions` is always the same.  Why not inline this block of JSX into your return value, below?